File: src/Addins/Eto.Addin.XamarinStudio/Templates/Projects/VB/MainForm.vb
Imports System
Imports Eto.Forms
Imports Eto.Drawing

''' <summary>
''' Your application's main form
''' </summary>
Public Class MainForm
	Inherits Form

	Public Sub New()
		Title = "My Eto Form"
		ClientSize = New Size(400, 350)

		Dim helloLabel As New Label()
		helloLabel.Text = "Hello World!"

		Dim layout As New StackLayout()
		layout.Items.Add(New StackLayoutItem(helloLabel))
		' Add more controls here

		Content = layout

		' create a few commands that can be used for the menu and toolbar
		Dim clickMe As New Command()
		With clickMe
			.MenuText = "Click Me!"
			.ToolBarText = "Click Me!"
			AddHandler .Executed, AddressOf ClickMeClicked
		End With

		Dim quitCommand As New Command()
		With quitCommand
			.MenuText = "Quit"
			.Shortcut = Application.Instance.CommonModifier Or Keys.Q
			AddHandler .Executed, AddressOf QuitClicked
		End With

		Dim aboutCommand As New Command()
		With aboutCommand
			.MenuText = "About..."
			AddHandler .Executed, AddressOf AboutClicked
		End With


		' create menu
			' File submenu
				' new ButtonMenuItem { Text = "&Edit", Items = { /* commands/items */ } },
				' new ButtonMenuItem { Text = "&View", Items = { /* commands/items */ } },
			' application (OS X) or file menu (others)
		Menu = New MenuBar()
		With Menu
			Dim fileItem As New ButtonMenuItem()
			With fileItem
				.Text = "&File"
				.Items.Add(clickMe)
			End With
			.Items.Add(fileItem)

			Dim preferencesItem As New ButtonMenuItem()
			preferencesItem.Text = "&Preferences..."
			.ApplicationItems.Add(preferencesItem)
			.QuitItem = quitCommand
			.AboutItem = aboutCommand
		End With

		' create toolbar			
		ToolBar = New ToolBar()
		ToolBar.Items.Add(clickMe)
	End Sub

	Sub QuitClicked(ByVal sender As Object, ByVal e As EventArgs)
		Application.Instance.Quit()
	End Sub

	Sub AboutClicked(ByVal sender As Object, ByVal e As EventArgs)
		MessageBox.Show(Me, "About my app...")
	End Sub

	Sub ClickMeClicked(ByVal sender As Object, ByVal e As EventArgs)
		MessageBox.Show(Me, "I was clicked!")
	End Sub
End Class

File: src/Addins/Eto.Addin.XamarinStudio/Templates/Projects/CS/MainForm.cs
using System;
using Eto.Forms;
using Eto.Drawing;

namespace ${Namespace}
{
	/// <summary>
	/// Your application's main form
	/// </summary>
	public $if$($UseCodePreview$==True)partial$endif$ class MainForm : Form
	{
		$if$($UseCodePreview$==True)void InitializeComponent()$endif$$if$($UseCode$==True)public MainForm()$endif$
		{
			Title = "My Eto Form";
			ClientSize = new Size(400, 350);

			Content = new StackLayout
			{
				Items =
				{
					"Hello World!",
					// Add more controls here
				}
			};

			// create a few commands that can be used for the menu and toolbar
			var clickMe = new Command { MenuText = "Click Me!", ToolBarText = "Click Me!" };
			clickMe.Executed += (sender, e) => MessageBox.Show(this, "I was clicked!");

			var quitCommand = new Command { MenuText = "Quit", Shortcut = Application.Instance.CommonModifier | Keys.Q };
			quitCommand.Executed += (sender, e) => Application.Instance.Quit();

			var aboutCommand = new Command { MenuText = "About..." };
			aboutCommand.Executed += (sender, e) => MessageBox.Show(this, "About my app...");

			// create menu
			Menu = new MenuBar
			{
				Items = {
					// File submenu
					new ButtonMenuItem { Text = "&File", Items = { clickMe } },
					// new ButtonMenuItem { Text = "&Edit", Items = { /* commands/items */ } },
					// new ButtonMenuItem { Text = "&View", Items = { /* commands/items */ } },
				},
				ApplicationItems = {
					// application (OS X) or file menu (others)
					new ButtonMenuItem { Text = "&Preferences..." },
				},
				QuitItem = quitCommand,
				AboutItem = aboutCommand
			};

			// create toolbar			
			ToolBar = new ToolBar { Items = { clickMe } };
		}
	}
}